UART-Between-Boards/code.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: Copyright (c) 2021 Dylan Herrada for Adafruit Industries
+#
+# SPDX-License-Identifier: Unlicense
+"""
+Code for communicating between two CircuitPython boards using UART.
+Increments built-in neopixel values of the other board starting from blue, then green and red.
+"""
+import time
+import board
+import busio
+import neopixel
+
+pixel = neopixel.NeoPixel(board.NEOPIXEL, 1, brightness=1.0, auto_write=True)
+
+uart = busio.UART(board.TX, board.RX, baudrate=9600)
+
+
+array = [0x55, 0, 0, 0, 0xAA]
+print(array)
+barray = bytearray(array)
+print(barray)
+
+uart.write(barray)
+while True:
+    data = uart.read(32)  # read up to 32 bytes
+    if data is not None:
+        msg = []
+        print(data)
+        if data[0] == 85 and data[-1] == 170:
```

Comment:
I'd factor out 0x55 and 0xAA so that you have constants.

Reply:
This check will breakdown if you receive two packets with the read. Do you want to handle if the read occurs in the middle of a packet?

Reply:
@tannewt By factoring them out, you mean setting them to their own variables, right?

> This check will breakdown if you receive two packets with the read.

So that was a big issue before I added this block, but after I added that block I stopped having those issues. 

> Do you want to handle if the read occurs in the middle of a packet?

Wouldn't checking the first and last byte solve this issue?

Reply:
> @tannewt By factoring them out, you mean setting them to their own variables, right?

Yup, into a constant so that they are defined in one place.

> > This check will breakdown if you receive two packets with the read.
> 
> So that was a big issue before I added this block, but after I added that block I stopped having those issues.

I think you'd want to check the overall length as well. By just checking start and end, then you'll miss the start and end in the middle of the two packets.

> > Do you want to handle if the read occurs in the middle of a packet?
> 
> Wouldn't checking the first and last byte solve this issue?

It'll drop partial packets so if you miss one, then it might be hard to get back aligned and you'll keep dropping them.

Reply:
> I think you'd want to check the overall length as well. By just checking start and end, then you'll miss the start and end in the middle of the two packets.

Ah ok. That's a good way to do it. 

> It'll drop partial packets so if you miss one, then it might be hard to get back aligned and you'll keep dropping them.

Good point. I think a good solution to this could be storing the last packet sent and resending it if that happens.

Reply:
You probably also want your read to align to your packet size. (So 5 instead of 32.) That way busio.UART can buffer any more incoming packets while you process the most recent one.

When you end up in the middle of the packet you need to read byte-by-byte until you find your start value.

Don't get to the point of resending. Reliable transmission is way harder than just noticing a packet was dropped. (TCP/IP does reliable transmission and it's not simple.)